src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -117,10 +116,14 @@ public sealed override ImmutableArray<string> FixableDiagnosticIds
         /// <param name="syntaxFacts">The syntax facts for the current language.</param>
         /// <returns>The replacement node to use in the rewritten syntax tree; otherwise, <see langword="null"/> to only
         /// rewrite the node originally rewritten by <see cref="TryUpdateNameForFlaggedNode"/>.</returns>
-        protected virtual SyntaxNode? TryUpdateParentOfUpdatedNode(SyntaxNode parent, SyntaxNode newNameNode, SyntaxEditor editor, ISyntaxFacts syntaxFacts)
-        {
-            return null;
-        }
+        protected virtual SyntaxNode? TryUpdateParentOfUpdatedNode(SyntaxNode parent, SyntaxNode newNameNode, SyntaxEditor editor, ISyntaxFacts syntaxFacts) => null;
+
+        /// <summary>
+        /// For C# we need to "unwrap" GlobalStatementSyntax in order to properly detect if it is a foreach statement. See https://github.com/dotnet/roslyn/issues/60030
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        protected virtual SyntaxNode UnwrapGlobalStatementIfNecessary(SyntaxNode node) => node;
```

Comment:
in general prefer making these abstract so that both languages have to consider and doc what they're doing.  it's fine for one language to override and then explicitly say "we have nothign to do here".